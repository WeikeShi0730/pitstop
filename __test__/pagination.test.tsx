import { render, RenderResult } from "@testing-library/react";
import Pagination from "../components/pagination.component";

let documentBody: RenderResult;
describe("<Pagination />", () => {
  beforeEach(
    () =>
      (documentBody = render(
        <Pagination
          setCurrentPage={() => {}}
          numPages={10}
          currentPage={1}
          scroll={0}
        />
      ))
  );
  it("Test <Pagination />", () => {
    expect(documentBody.queryByText("0")).not.toBeInTheDocument();
    expect(documentBody.queryByText("1")).toBeInTheDocument();
    expect(documentBody.queryByText("10")).toBeInTheDocument();
    expect(documentBody.queryByText("11")).not.toBeInTheDocument();
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
