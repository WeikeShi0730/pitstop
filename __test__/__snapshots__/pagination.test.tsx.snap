// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> Test <Pagination /> 1`] = `
<body>
  <div>
    <div>
      <div
        aria-orientation="horizontal"
        class="flex w-fit max-w-full p-1 gap-x-1 bg-slate-300 rounded-lg shadow text-sm md:text-base"
        role="tablist"
      >
        <button
          class="buttonPrev w-8 py-1 font-light leading-5 text-slate-700 rounded-lg hover:bg-slate-100 text-slate-400 hover:bg-transparent"
          disabled=""
        >
          &lt;
        </button>
        <button
          aria-selected="true"
          class="font-light w-8 py-1 leading-5 text-slate-700 rounded-lg focus:outline-none transition-all ease-in-out duration-200 bg-orange-theme shadow text-slate-100"
          id="headlessui-tabs-tab-1"
          role="tab"
          tabindex="0"
          type="button"
        >
          1
        </button>
        <button
          aria-selected="false"
          class="font-light w-8 py-1 leading-5 text-slate-700 rounded-lg focus:outline-none transition-all ease-in-out duration-200 hover:bg-slate-100 text-slate-700"
          id="headlessui-tabs-tab-2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          2
        </button>
        <button
          aria-selected="false"
          class="font-light w-8 py-1 leading-5 text-slate-700 rounded-lg focus:outline-none transition-all ease-in-out duration-200 hover:bg-slate-100 text-slate-700"
          id="headlessui-tabs-tab-3"
          role="tab"
          tabindex="-1"
          type="button"
        >
          3
        </button>
        <button
          aria-selected="false"
          class="font-light w-8 py-1 leading-5 text-slate-700 rounded-lg focus:outline-none transition-all ease-in-out duration-200 hover:bg-slate-100 text-slate-700"
          id="headlessui-tabs-tab-4"
          role="tab"
          tabindex="-1"
          type="button"
        >
          4
        </button>
        <button
          aria-selected="false"
          class="font-light w-8 py-1 leading-5 text-slate-700 rounded-lg focus:outline-none transition-all ease-in-out duration-200 hover:bg-slate-100 text-slate-700"
          id="headlessui-tabs-tab-5"
          role="tab"
          tabindex="-1"
          type="button"
        >
          5
        </button>
        <button
          aria-selected="false"
          class="font-light w-8 py-1 leading-5 text-slate-700 rounded-lg focus:outline-none transition-all ease-in-out duration-200 hover:bg-slate-100 text-slate-700"
          id="headlessui-tabs-tab-6"
          role="tab"
          tabindex="-1"
          type="button"
        >
          6
        </button>
        <button
          aria-selected="false"
          class="font-light w-8 py-1 leading-5 text-slate-700 rounded-lg focus:outline-none transition-all ease-in-out duration-200 hover:bg-slate-100 text-slate-700"
          id="headlessui-tabs-tab-7"
          role="tab"
          tabindex="-1"
          type="button"
        >
          7
        </button>
        <button
          aria-selected="false"
          class="font-light w-8 py-1 leading-5 text-slate-700 rounded-lg focus:outline-none transition-all ease-in-out duration-200 hover:bg-slate-100 text-slate-700"
          id="headlessui-tabs-tab-8"
          role="tab"
          tabindex="-1"
          type="button"
        >
          8
        </button>
        <button
          aria-selected="false"
          class="font-light w-8 py-1 leading-5 text-slate-700 rounded-lg focus:outline-none transition-all ease-in-out duration-200 hover:bg-slate-100 text-slate-700"
          id="headlessui-tabs-tab-9"
          role="tab"
          tabindex="-1"
          type="button"
        >
          9
        </button>
        <button
          aria-selected="false"
          class="font-light w-8 py-1 leading-5 text-slate-700 rounded-lg focus:outline-none transition-all ease-in-out duration-200 hover:bg-slate-100 text-slate-700"
          id="headlessui-tabs-tab-10"
          role="tab"
          tabindex="-1"
          type="button"
        >
          10
        </button>
        <button
          class="buttonNext w-8 py-1 font-light leading-5 text-slate-700 rounded-lg hover:bg-slate-100"
        >
          &gt;
        </button>
      </div>
    </div>
  </div>
</body>
`;
